one/src/daemon.rs

```diff
@@ -323,11 +323,33 @@ fn spawn_database_optimizer(
     mut shutdown: tokio::sync::broadcast::Receiver<()>,
 ) -> tokio::task::JoinHandle<()> {
     tokio::spawn(async move {
+        // Recommended practice is that applications with long-lived database connections should run "PRAGMA optimize=0x10002"
+        // when the database connection first opens, then run "PRAGMA optimize" again at periodic intervals - perhaps once per day
+        // https://www.sqlite.org/pragma.html#pragma_optimize
+        match sqlite_pool.run_statement("PRAGMA optimize=0x10002").await {
+            Ok(_) => {
+                info!("successfully ran sqlite optimize=0x10002");
+            }
+            Err(e) => {
+                warn!(error=?e, "failed to run initial database optimize statement");
+            }
+        }
+        let mut duration = std::time::Duration::from_secs(60 * 60 * 24); // once daily
         loop {
-            let mut duration = std::time::Duration::from_secs(60 * 60 * 24); // once daily
+            let mut interval = tokio::time::interval(duration);
+            interval.tick().await; // first tick is immediate
```

Comment:
Why do we manually run optimize above and then throw away the first tick? Also why do we recreate the interval within the loop. I would expect to create the interval once and then let it fire immediately to do the first optimization run. But maybe there is a difference between the optimization runs I do not understand.

Reply:
yeah, I did it a bit weird didn't I? I changed it so we don't do a first run in the task anymore. It was hanging forever on sqlx 0.8 and preventing the ordering task from getting a writer connection. Now it does one pass in the daemon inline and spawns this task if `analysis_limit` is configured (by default yes). It resets the interval in case we changed how long it should be due to an error, but a sleep or alternative might be better.

Reply:
Agreed, lets just use a sleep since we never read more than two ticks from the interval and we have a dynamic period we want to wait.